packages/server/src/workers/reindex.ts

```diff
@@ -174,39 +179,51 @@ async function processPage(searchRequest: SearchRequest, job: Job<ReindexJobData
       duration: `${elapsedTime} ms`,
     });
 
-    job.data.results[resourceType] = [newCount, elapsedTime];
-    await updateStatus(job);
-
-    // Enqueue job to start reindexing the next resource type
-    await addReindexJobData({
-      ...job.data,
-      resourceTypes: resourceTypes.slice(1),
-      currentTimestamp: new Date(0).toISOString(),
-      count: 0,
-      startTime: Date.now(),
-    });
+    return { count: newCount, duration: elapsedTime };
   } else {
-    await finishReindex(job, newCount);
+    const elapsedTime = Date.now() - job.data.startTime;
+    getRequestContext().logger.info('Reindex completed', { resourceType, count, duration: `${elapsedTime} ms` });
+    return { count: newCount, duration: elapsedTime };
   }
 }
 
-async function updateStatus(job: Job<ReindexJobData>): Promise<void> {
+function searchRequestForNextPage(job: Job<ReindexJobData>): SearchRequest {
+  const { resourceTypes, currentTimestamp, endTimestamp, searchFilter } = job.data;
+  const resourceType = resourceTypes[0];
+  const searchRequest: SearchRequest = {
+    resourceType,
+    count: batchSize,
+    sortRules: [{ code: '_lastUpdated', descending: false }],
+    filters: [
```

Comment:
For consideration:  With the new `_cursor` feature, it will automatically handle this filtering and pagination.  It would have the added benefit of being more robust with resources with identical `meta.lastUpdated`, and would avoid double indexing the resources at page boundaries.

It should be a matter of replacing `currentTimestamp` with `cursor`.

See the bulk data PR for a reference: https://github.com/medplum/medplum/pull/5102